pkg/util/typeutil/schema.go

```diff
@@ -1458,7 +1458,7 @@ func GetPK(data *schemapb.IDs, idx int64) interface{} {
 }
 
 func GetDataIterator(field *schemapb.FieldData) func(int) any {
-	if field.ValidData != nil {
+	if field.GetValidData() != nil {
 		// unpack valid data
```

Comment:
old code will crash here, as field is nil for no entities returned by querynode
this will never happen after this fix